diff --git a/./upstream/emscripten/src/worker.orig.js b/./upstream/emscripten/src/worker.js
index 6bee012..22bfb02 100644
--- a/./upstream/emscripten/src/worker.orig.js
+++ b/./upstream/emscripten/src/worker.js
@@ -197,7 +197,7 @@ function handleMessage(e) {
 #endif
 
 #if MODULARIZE && EXPORT_ES6
-      (e.data.urlOrBlob ? import(e.data.urlOrBlob) : import('./{{{ TARGET_JS_NAME }}}'))
+      (import('./{{{ TARGET_JS_NAME }}}'))
       .then(exports => exports.default(Module));
 #else
       if (typeof e.data.urlOrBlob == 'string') {
