diff --git a/./upstream/emscripten/tools/file_packager.orig.py b/./upstream/emscripten/tools/file_packager.py
index 956ab8a..f3df191 100755
--- a/./upstream/emscripten/tools/file_packager.orig.py
+++ b/./upstream/emscripten/tools/file_packager.py
@@ -739,7 +739,7 @@ def generate_js(data_target, data_files, metadata):
             DataRequest.prototype.requests[files[i].filename].onload();
           }'''
       use_data += ("          Module['removeRunDependency']('datafile_%s');\n"
-                   % js_manipulation.escape_for_js_string(data_target))
+                   % js_manipulation.escape_for_js_string(os.path.basename(data_target)))
 
     else:
       # LZ4FS usage
@@ -753,7 +753,7 @@ def generate_js(data_target, data_files, metadata):
             compressedData['data'] = byteArray;
             assert(typeof Module['LZ4'] === 'object', 'LZ4 not present - was your app build with -sLZ4?');
             Module['LZ4'].loadPackage({ 'metadata': metadata, 'compressedData': compressedData }, %s);
-            Module['removeRunDependency']('datafile_%s');''' % (meta, "true" if options.use_preload_plugins else "false", js_manipulation.escape_for_js_string(data_target))
+            Module['removeRunDependency']('datafile_%s');''' % (meta, "true" if options.use_preload_plugins else "false", js_manipulation.escape_for_js_string(os.path.basename(data_target)))
 
     package_name = data_target
     remote_package_size = os.path.getsize(package_name)
@@ -772,7 +772,7 @@ def generate_js(data_target, data_files, metadata):
         Module['locateFile'] = Module['locateFilePackage'];
         err('warning: you defined Module.locateFilePackage, that has been renamed to Module.locateFile (using your locateFilePackage for now)');
       }
-      var REMOTE_PACKAGE_NAME = Module['locateFile'] ? Module['locateFile'](REMOTE_PACKAGE_BASE, '') : REMOTE_PACKAGE_BASE;\n''' % (js_manipulation.escape_for_js_string(data_target), js_manipulation.escape_for_js_string(remote_package_name))
+      var REMOTE_PACKAGE_NAME = Module['locateFile'] ? Module['locateFile'](REMOTE_PACKAGE_BASE, '') : REMOTE_PACKAGE_BASE;\n''' % (js_manipulation.escape_for_js_string(remote_package_name), js_manipulation.escape_for_js_string(remote_package_name))
     metadata['remote_package_size'] = remote_package_size
     ret += '''var REMOTE_PACKAGE_SIZE = metadata['remote_package_size'];\n'''
 
@@ -1022,7 +1022,7 @@ def generate_js(data_target, data_files, metadata):
         var curr;
         %s
       };
-      Module['addRunDependency']('datafile_%s');\n''' % (use_data, js_manipulation.escape_for_js_string(data_target))
+      Module['addRunDependency']('datafile_%s');\n''' % (use_data, js_manipulation.escape_for_js_string(remote_package_name))
     # use basename because from the browser's point of view,
     # we need to find the datafile in the same dir as the html file
 
