diff --git a/./upstream/emscripten/emcc.py.orig b/./upstream/emscripten/emcc.py
index dfe088e..7c85358 100644
--- a/./upstream/emscripten/emcc.py.orig
+++ b/./upstream/emscripten/emcc.py
@@ -385,14 +385,8 @@ def get_clang_flags(user_args):
     if '-mbulk-memory' not in user_args:
       flags.append('-mbulk-memory')
 
-  # In emscripten we currently disable bulk memory by default.
-  # This should be removed/updated when we als update the default browser targets.
-  if '-mbulk-memory' not in user_args and '-mno-bulk-memory' not in user_args:
-    # Bulk memory may be enabled via threads or directly via -s.
-    if not settings.BULK_MEMORY:
-      flags.append('-mno-bulk-memory')
-  if '-mnontrapping-fptoint' not in user_args and '-mno-nontrapping-fptoint' not in user_args:
-    flags.append('-mno-nontrapping-fptoint')
+  flags.append('-mbulk-memory')
+  flags.append('-matomics')
 
   if settings.RELOCATABLE and '-fPIC' not in user_args:
     flags.append('-fPIC')
