diff --git a/./upstream/emscripten/tools/file_packager.orig.py b/./upstream/emscripten/tools/file_packager.py

--- emsdk/upstream/emscripten/tools/file_packager.orig.py
+++ emsdk/upstream/emscripten/tools/file_packager.py
@@ -768,12 +768,11 @@
         PACKAGE_PATH = encodeURIComponent(location.pathname.toString().substring(0, location.pathname.toString().lastIndexOf('/')) + '/');
       }
       var PACKAGE_NAME = '%s';
-      var REMOTE_PACKAGE_BASE = '%s';
       if (typeof Module['locateFilePackage'] === 'function' && !Module['locateFile']) {
         Module['locateFile'] = Module['locateFilePackage'];
         err('warning: you defined Module.locateFilePackage, that has been renamed to Module.locateFile (using your locateFilePackage for now)');
       }
-      var REMOTE_PACKAGE_NAME = Module['locateFile'] ? Module['locateFile'](REMOTE_PACKAGE_BASE, '') : REMOTE_PACKAGE_BASE;\n''' % (js_manipulation.escape_for_js_string(remote_package_name), js_manipulation.escape_for_js_string(remote_package_name))
+      var REMOTE_PACKAGE_NAME = new URL('./%s', import.meta.url).href;\n''' % (js_manipulation.escape_for_js_string(remote_package_name), js_manipulation.escape_for_js_string(remote_package_name))
     metadata['remote_package_size'] = remote_package_size
     ret += '''var REMOTE_PACKAGE_SIZE = metadata['remote_package_size'];\n'''
 
@@ -1112,7 +1111,7 @@
 
   function runMetaWithFS() {
     Module['addRunDependency']('%(metadata_file)s');
-    var REMOTE_METADATA_NAME = Module['locateFile'] ? Module['locateFile']('%(metadata_file)s', '') : '%(metadata_file)s';
+    var REMOTE_METADATA_NAME = new URL('./%(metadata_file)s', import.meta.url).href;
     var xhr = new XMLHttpRequest();
     xhr.onreadystatechange = function() {
      if (xhr.readyState === 4 && xhr.status === 200) {
